### PR TITLE
Sparky2: switch to ChibiOS

### PR DESCRIPTION
**Pull request by fujin**
(cherry picked from commit 9b47395832b7aced528eb602f2325a97dee7d2ef)
Signed-off-by: AJ Christensen <aj@junglistheavy.industries>

### CODE REVIEW

**Comment by fujin** [comment]
Grabbed from https://github.com/peabody124/TauLabs/commits/sparky2_ins14

File: flight/targets/sparky2/fw/board.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+/*
+ * Setup for STMicroelectronics STM32F4-Discovery board.
+ */
+
+/*
+ * Board identifier.
+ */
+#define BOARD_QUANTEC_QUANTON
+#define BOARD_NAME                  "quanton flight control"
```
Comment (tracernz) [inline flight/targets/sparky2/fw/board.h]: Intentional?

Comment (fujin) [inline flight/targets/sparky2/fw/board.h]: I can update it. I copied the commit as-is from @peabody124's branch. When I did the ChibiOS/RT upgrade for BrainFPV, it was named similarly, and the comment on line 23 was regarding a Discovery board too. I updated the Brain one.

Preferences? I can easily update!

Comment (peabody124) [inline flight/targets/sparky2/fw/board.h]: Yeah that should be accurate :) my bad!

**Comment by peabody124** [comment]
I'd like to get some reports of flight testing with this and also if you get reboots when plugging into USB. I had an incident when the motors briefly spooled up on that reboot which is why I've hesitated to push this. I had a number of other changes going on at the time so could not attribute it entirely to ChiBios. Please test without props for a bit first!

File: flight/targets/sparky2/fw/board.h (hunk 1 above)
Comment (fujin) [inline flight/targets/sparky2/fw/board.h]: updated in 77b8e5a

**Comment by fujin** [comment]
@peabody124 will do :-) :recycle: :tea:

**Comment by BrainFPV** [comment]
@peabody124 FYI, @fujin was nice enough to also move the Brain to ChibiOS and so far I haven't noticed any problems, so I recently switched to ChibiOS for the Brain. Good stuff :)

**Comment by solidgoldbomb** [comment]
jenkins: test this please

**Comment by peabody124** [comment]
I'm good for this to merge after the release is done.

**Pull request merged**